scripts/build.sh

```diff
@@ -189,6 +212,15 @@ if $generate; then
   fi
 fi
 
+if $documentation; then
+  Generate
+  GenerateDocumentationOnly
+
+  if [ "$LASTEXITCODE" != 0 ]; then
+    return "$LASTEXITCODE"
+  fi
+fi
+
```

Comment:
If the documentation flag is passed, it assumes that Generate was never called to generate the Doxyfile so we call it manually here.

Reply:
Why not just `generate=true` if `documentation` is `true`, just like we do for `CI` so it flows naturally and works with the rest of the script?

Reply:
I think it's really just an oversight I had from the first implementation - the thought was to not build since it was unnecessary, we only needed to generate and then build the `Doxygen` target as needed.

I'll correct this

Reply:
Rewrote this to follow the general flow for CI